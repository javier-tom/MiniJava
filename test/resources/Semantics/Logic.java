class Main {
    public static void main(String[] args) {
        System.out.println((new Vars().run(10, 15)));
    }
}
class Vars {
    int a;
    int b;
    int c;
    boolean f1;
    boolean f2;

    public int run(int x, int y) { // x = 10, y = 15
        if (x < y) { // true
            f1 = this.run2(a, y);
        } else {
            arr = this.run3(true,false); // should not run
        }
        return 1;
    }

    public boolean run2(int a, int b) { // a = 0?, b = 15
        int[] arr;
        while (a < b) {
            a = a + 1;
        }
        arr = this.run3(a, b); // a = 15, b = 15
        return true;
    }

    public int[] run3(int a, int b) {
        int[] arr;
        arr = new int[a];
        a = a - b; // a = 0
        while(a < b - 10) { // ?
            arr[a] = 1;
            a = a + 1;
        }
        return arr;
    }
}