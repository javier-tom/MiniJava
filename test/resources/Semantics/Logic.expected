class Vars:
  super: null
  fields:
    int a;
    int b;
    int c;
    boolean f1;
    boolean f2;
  method run2:
    params: int, int,
    returns: boolean
    locals:
      int[] arr;
  method run:
    params: int, int,
    returns: int
    locals:
  method run3:
    params: int, int,
    returns: int[]
    locals:
      int[] arr;

Error on line 17: unknown variable arr
Error on line 17: expected type int in parameter 1 of the call to run3. Got type boolean instead.
Error on line 17: expected type int in parameter 2 of the call to run3. Got type boolean instead.
Error on line 17: cannot assign int[] to a *error